From 6e65a4d2971b317796e3b71764af5f83b0b93fc8 Mon Sep 17 00:00:00 2001
From: Brad Geltz <brad.geltz@intel.com>
Date: Thu, 2 Aug 2018 11:34:34 -0700
Subject: [PATCH 3/3] Increase problem size for Theta.

- Increase iterations (iter) to 2000.  Nek5k might run for 12 hours
  after a few minutes in setup.
- Increase lp and lelt to 262144 and 6200, respectively.
  + lp calculation: This is the maximum number of nodes that could
  possibly be utilized multiplied by the number of CPU cores on each
  node.  On Theta that's 4096 nodes * 64 cores = 262144.
  + lelt calculation: It's recommended to have 100 elements per core on
  KNL.  The number of cores/threads desired impacts the elements per
  core calculation.  On a 64 core system where 1 core is left free for
  the OS, and 1 core is left free for the GEOPM Controller, 62 cores
  remain.  62 * 100 = 6200.
- Set poly. order range to only solve 12th order case.  Nek5k would only
  solve a single order.
- Nekbone is not sensitive to the node count, but it's generally best if
  the rank count can be factored into three numbers.
- Information obtained from nekbone maintainer at Argonne: sparker <at>
  anl <dot> gov.

#  Copyright (c) 2015, 2016, 2017, 2018, Intel Corporation
#
#  Redistribution and use in source and binary forms, with or without
#  modification, are permitted provided that the following conditions
#  are met:
#
#      * Redistributions of source code must retain the above copyright
#        notice, this list of conditions and the following disclaimer.
#
#      * Redistributions in binary form must reproduce the above copyright
#        notice, this list of conditions and the following disclaimer in
#        the documentation and/or other materials provided with the
#        distribution.
#
#      * Neither the name of Intel Corporation nor the names of its
#        contributors may be used to endorse or promote products derived
#        from this software without specific prior written permission.
#
#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
#

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
---
 src/driver.f           | 2 +-
 test/example1/SIZE     | 4 ++--
 test/example1/data.rea | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/driver.f b/src/driver.f
index 76b5493..04d511d 100644
--- a/src/driver.f
+++ b/src/driver.f
@@ -67,7 +67,7 @@ c     SET UP and RUN NEKBONE
 
            call proxy_setup(ah,bh,ch,dh,zpts,wght,g) 
 
-           niter = 100
+           niter = 2000
            n     = nx1*ny1*nz1*nelt
 
            call set_f(f,c,n)
diff --git a/test/example1/SIZE b/test/example1/SIZE
index 941aedc..8223258 100644
--- a/test/example1/SIZE
+++ b/test/example1/SIZE
@@ -3,8 +3,8 @@ C     Dimension file to be included
       parameter (ldim=3)                      ! dimension
       parameter (lx1=12,ly1=lx1,lz1=lx1)      ! polynomial order
 
-      parameter (lp =49152)                 ! max number of processors
-      parameter (lelt= 512)                    ! max number of elements, per proc
+      parameter (lp = 262144)                 ! max number of processors
+      parameter (lelt= 6200)                    ! max number of elements, per proc
 
       parameter (lelg=lelt*lp)                ! max total elements in a test
       parameter (lelx=lelg,lely=1,lelz=1)     ! max elements in each direction
diff --git a/test/example1/data.rea b/test/example1/data.rea
index 36077f4..cc1c6a7 100644
--- a/test/example1/data.rea
+++ b/test/example1/data.rea
@@ -1,5 +1,5 @@
 .true. = ifbrick               ! brick or linear geometry
 512 512 1  = iel0,ielN,ielD (per processor)  ! range of number of elements per proc.
- 9  12 3 = nx0,nxN,nxD         ! poly. order range for nx1
+ 12 12 1 = nx0,nxN,nxD         ! poly. order range for nx1
  1  1  1 = npx, npy, npz       ! processor distribution in x,y,z
  1  1  1 = mx, my, mz          ! local element distribution in x,y,z
-- 
2.12.3

